packages/react-server/core/renderMiddleware.js

```diff
@@ -880,6 +881,13 @@ function writeElement(res, element, i){
 	}
 }
 
+function logAboveTheFoldTime(res) {
+	// write a synchronous script to record the time on the browser when above the fold content shows up
+	// this is a proxy for "first paint" when the DOM is parsed and painted
+	renderScriptsSync([{text:`__displayAboveTheFold=new Date;
+		window.performance && window.performance.mark && window.performance.mark("displayAboveTheFold.fromStart");`}], res);
```

Comment:
Whoa, this is cool!  Didn't even know about [`performance.mark`](https://developer.mozilla.org/en-US/docs/Web/API/Performance/mark).

I appreciate this being broken onto two lines from a readability perspective, but because this is an es6 template string it will insert a literal newline and tabs into the output document.

This might be a good place for an old-fashioned string concatenation:
```javascript
[{text:'__displayAboveTheFold=new Date;'+
'window.performance && ...'}],